controllers/apim/ratelimitpolicy_controller_test.go

```diff
@@ -0,0 +1,269 @@
+					Protocol: gatewayapiv1alpha2.ProtocolType("HTTP"),
+				},
+			},
+		},
+	}
+}
+
+func testBuildBasicHttpRoute(routeName, gwName, ns string) *gatewayapiv1alpha2.HTTPRoute {
+	tmpMatchPathPrefix := gatewayapiv1alpha2.PathMatchPathPrefix
+	tmpMatchValue := "/toy"
+	tmpMatchMethod := gatewayapiv1alpha2.HTTPMethod("GET")
+
+	return &gatewayapiv1alpha2.HTTPRoute{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "HTTPRoute",
+			APIVersion: gatewayapiv1alpha2.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      routeName,
+			Namespace: ns,
+			Labels:    map[string]string{"app": "rlptest"},
+		},
+		Spec: gatewayapiv1alpha2.HTTPRouteSpec{
+			CommonRouteSpec: gatewayapiv1alpha2.CommonRouteSpec{
+				ParentRefs: []gatewayapiv1alpha2.ParentRef{
+					{
+						Name: gatewayapiv1alpha2.ObjectName(gwName),
+					},
+				},
+			},
+			Hostnames: []gatewayapiv1alpha2.Hostname{"*.example.com"},
+			Rules: []gatewayapiv1alpha2.HTTPRouteRule{
+				{
+					Matches: []gatewayapiv1alpha2.HTTPRouteMatch{
+						{
+							Path: &gatewayapiv1alpha2.HTTPPathMatch{
+								Type:  &tmpMatchPathPrefix,
+								Value: &tmpMatchValue,
+							},
+							Method: &tmpMatchMethod,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func testBuildBasicRoutePolicy(policyName, ns, routeName string) *apimv1alpha1.RateLimitPolicy {
+	genericDescriptorKey := "op"
+
+	return &apimv1alpha1.RateLimitPolicy{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "RateLimitPolicy",
+			APIVersion: apimv1alpha1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      policyName,
+			Namespace: ns,
+		},
+		Spec: apimv1alpha1.RateLimitPolicySpec{
+			TargetRef: gatewayapiv1alpha2.PolicyTargetReference{
+				Group: gatewayapiv1alpha2.Group("gateway.networking.k8s.io"),
+				Kind:  "HTTPRoute",
+				Name:  gatewayapiv1alpha2.ObjectName(routeName),
+			},
+			RateLimits: []apimv1alpha1.RateLimit{
+				{
+					Configurations: []apimv1alpha1.Configuration{
+						{
+							Actions: []apimv1alpha1.ActionSpecifier{
+								{
+									GenericKey: &apimv1alpha1.GenericKeySpec{
+										DescriptorValue: "1",
+										DescriptorKey:   &genericDescriptorKey,
+									},
+								},
+							},
+						},
+					},
+					Limits: []apimv1alpha1.Limit{
+						{
+							MaxValue:   5,
+							Seconds:    10,
+							Conditions: []string{"op == 1"},
+							Variables:  []string{},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+var _ = Describe("RateLimitPolicy controller", func() {
+	var (
+		testNamespace        string
+		genericDescriptorKey string = "op"
+	)
+
+	BeforeEach(CreateNamespaceCallback(&testNamespace))
+
+	AfterEach(DeleteNamespaceCallback(&testNamespace))
+
+	Context("Basic: RLP targeting HTTPRoute", func() {
+		It("check created resources", func() {
+
+			routeName := "toystore-route"
+			gwName := "toystore-gw"
+			rlpName := "toystore-rlp"
+
+			gateway := testBuildBasicGateway(gwName, testNamespace)
```

Comment:
Maybe there's a way to have _Factories_ for these mocked instances, so we could reuse them in other tests and build them with specific params? Something similar to "FactoryGirl" in Ruby maybe ?

Reply:
they can be reused from all the tests in the same package. 

But, yeah.. we can do much better. maybe in another PR?